### PR TITLE
Userpages - more dynamic debugging

### PR DESCRIPTION
**Pull request by MariagraziaAlastra**
//cc @jagtalon @jbarrett Almost done!

### CODE REVIEW

**Comment by MariagraziaAlastra** [comment]
@bsstoner Ready for review :smile:

File: root/static/js/userpages/UserPage.js [hunk 1]
```diff
@@ -86,6 +103,16 @@ app.controller('UserPageController', function($scope, $http, fn) {
 
 	// by default. for 'filterable'
 	$scope.show_ias = ($scope.count.maintained_ias) ? $scope.maintained : $scope.ias_developed_only;
+
+    $scope.changeShownIAs = function(which) {
+        $scope.show_ias = which;
+
+        var objKey = _.findKey($scope, which);
+        console.log(objKey);
+        console.log($scope);
```
Comment (bsstoner) [inline root/static/js/userpages/UserPage.js]: did you want to leave these log statements in?

File: root/static/js/userpages/UserPage.js [hunk 2]
```diff
@@ -25,6 +26,22 @@ app.controller('UserPageController', function($scope, $http, fn) {
 	$scope.count = {};
 	$scope.topics = [];
 
+    $scope.imgUrl = 'https://duckduckgo.com/t/userpage_' + $scope.response.gh_data.id;
+    $scope.randomNum = Math.ceil(Math.random() * 1e7);
+    $scope.imgUrlSuffix = '?' + Math.ceil(Math.random() * 1e7);
+
+    $scope.newImg = new Image();
+    $scope.newImg.src = $scope.imgUrl + $scope.imgUrlSuffix;
```
Comment (bsstoner) [inline root/static/js/userpages/UserPage.js]: Could you just replace this with a call to `$scope.addImg($scope.response.gh_data.id)`?

File: root/static/js/userpages/UserPage.js (hunk 1 above)
Comment (MariagraziaAlastra) [inline root/static/js/userpages/UserPage.js]: @bsstoner Whoops, sorry, forgot about them. They're removed now!

**Comment by jagtalon** [comment]
@MariagraziaAlastra hi! Made a little change to c_issue.html due to https://app.asana.com/0/84322308542789/109571132613869.

**Comment by MariagraziaAlastra** [comment]
@jagtalon Np, fixed all the merge conflicts :smile:
@bsstoner now I'm stripping any non alphanumeric chars as well!

File: root/static/js/userpages/UserPage.js [hunk 3]
```diff
@@ -25,6 +26,23 @@ app.controller('UserPageController', function($scope, $http, fn) {
...
+
+    $scope.addImg = function(pre_id, element_id) {
+        $scope.randomNum = Math.ceil(Math.random() * 1e7);
+        pre_id = pre_id? '_' + pre_id : '';
+        element_id = element_id? '_' + element_id.replace(/[^0-9A-Za-z]/g, '') : '';
+        $scope.imgUrlSuffix = pre_id + element_id + '?' + $scope.randomNum;
```
Comment (bsstoner) [inline root/static/js/userpages/UserPage.js]: Why does `randomNum` need to be on `$scope`? It shouldn't be used outside the scope of this function, right?

Comment (MariagraziaAlastra) [inline root/static/js/userpages/UserPage.js]: @bsstoner Thanks, it's just a regular var now! :smile:

File: root/static/js/userpages/UserPage.js [hunk 4]
```diff
@@ -25,6 +26,22 @@ app.controller('UserPageController', function($scope, $http, fn) {
...
+    $scope.newImg.src = $scope.imgUrl + $scope.imgUrlSuffix;
+
+    $scope.addImg = function(pre_id, element_id) {
+        var randomNum = Math.ceil(Math.random() * 1e7);
+        pre_id = pre_id? '_' + pre_id : '';
+        element_id = element_id? '_' + element_id.replace(/[^0-9A-Za-z]/g, '') : '';
+        $scope.imgUrlSuffix = pre_id + element_id + '?' + randomNum;
+
+        $scope.newImg = new Image();
+        $scope.newImg.src = $scope.imgUrl + $scope.imgUrlSuffix;
```
Comment (bsstoner) [inline root/static/js/userpages/UserPage.js]: Seems like the same thing with `imgUrlSuffix`? That doesn't seem used anywhere either.

Comment (MariagraziaAlastra) [inline root/static/js/userpages/UserPage.js]: @bsstoner Right, removed that from $scope too!

**Comment by bsstoner** [comment]
I have no way of testing, but :+1: on the code as far as I can tell :)

**Comment by MariagraziaAlastra** [comment]
@bsstoner Thanks, I tested it again and looks like it works. Merging! :cactus:

**Pull request merged**